README.md

```diff
@@ -19,3 +20,9 @@ randomly without memory instead of systematically trying them?
 
 Describe your reasoning and the conclusion you've come to. Your reasoning is the
 most important part. Add your answer to this markdown file.
+
+## Answer
+
+The runtime complexity of 'permutationSort' is factorial or $O(N!)$, where $N$ is the length of the input array 'a'. The base case for the recursion is when 'currentIdx' equals the length of the input array 'arr'. The 'generatePermutations' function explores all possible permutations by swapping elements at the current index with each remaining element. For each 'currentIdx' there are $(N - currentIdx)$ possible choices for the next element swap, so each level of recursion has this time. The total number of recursive calls is the product of the branching factors at each level, this makes the time, $(N -0) * (N - 1) * (N - 2)$, which is $N!$. At each level of recursion the algorithm swaps elements and makes a call to 'generatePermutation', and each of these operations takes constant time. So the worst case is $O(N!)$, as each permutation needs to be generated and checked. The time complexity of checking whether a given permutation is sorted is $O(N)$ beacuse, in the worst case, the algorithm needs to iterate through all $N$ elements of the array. 
+
+The best case input for this algorithm would be an already sorted array where the check for sortedness would take $O(N)$ time. This means there would only need to be a single check with a time of $O(1)$. The worst case input would be an array that has all possible permutations of its elements, which would result in a time of $O(N! * N)$. If the permutations were generated randomly without memory, the complexity would still be the same, $O(N! * N)$. This is because the number of permutations to check remains the same. 
```

Comment:
This doesn't seem to be consistent anymore -- "This means there would only need to be a single check with a time of $O(1)$."